### PR TITLE
unstashable assembly bombs

### PR DESCRIPTION
**Pull request by mc-oofert**
## About The Pull Request
assembly bombs are bulky
(the like ones you make with a tank and an igniter and some other thing)

## Why It's Good For The Game

these things can (and usually do) gib people, are REALLY easy to massproduce and you can also fit several in a bag
while still easier to use to gib people than TTVs

at best i think making them bulky should prevent some dude from carrying like 8 bombs in his backpack and whipping them out and throwing without any warning because they would have to carry it (or invest in BoH but thats sci anyway so)

## Changelog
:cl:
balance: assemblybombs are bulky
/:cl:

### CODE REVIEW

**Comment by moocowswag** [comment]
baby steps please, its not a sophisticated mechanism like a tank transfer valve, its a single rigged tank, maybe some hissing noise or something to give away that its about to blow, alot of people are just as reliant on their ears as they are on their eyes and in their current state they behave alot like mute grenades.

**Review by ZephyrTFA — APPROVED** [review]

**Comment by OverwatchVoice** [comment]
I feel like the best solution to how easily abusable assembly bombs are is to get rid of the infrared trigger method, or somehow nerfing it. This would alleviate the meta way of using assembly bombs; That being essencially imact-triggered mini-bombs.

**Comment by Livrah** [comment]
The problem is how easy it is to make, not how easy it is to carry.
Nerf about 674 plasma which every assistant could make roundstart will affect good atmos players.

**Comment by Thunder12345** [comment]
> The problem is how easy it is to make, not how easy it is to carry. Nerf about 674 plasma which every assistant could make roundstart will affect good atmos players.

Nerfs based on making bomb recipes harder have a long and sordid history of working for about 2 days before the new optimal recipe becomes widely known.

**Comment by moocowswag** [comment]
I think livrah's point is that 674 is obtainable with round start thermo machines and nerfing it to a higher precise temperature would lock it behind upgraded machines.

**Comment by SomeoneYouProbablyKnow** [comment]
Making them bulky just turns them into bad ttvs. Yea carrying 8 of those is insane but.. that's our current storage.

If you really want to make them bulky consider letting people carey them in belt and suit slots with an obvious sprite. That way you can carry 2-3 and it's obvious when you do. A middle ground and not a redundancy.

**Comment by MrMelbert** [comment]
Would be cool if the temperature required you do a burn in the incinerator or something

**Comment by GuillaumePrata** [comment]
Making it bulky and easy to produce still keeps it as an emergency explosive you can easy make, which IMO is the best use case for it.

Malf AI just exposed herself? You can easily make a few explosives to breach in.
Heretic ascended? Crew with halos? You are the CE/RD and the revs are at your door? Perfect and easy explosive to use.

Now if we make it harder to achieve, that niche dies, while the boring "I spend 20 minutes at Atmos and then got out to give people" meta gets stronger, which is a really dumb thing to push for.

**Comment by Higgin** [comment]
Bulky is a good change at least until the actual ease and potency of these things also changes.

Presently they are not subject to the same material limits of TTVs (valves you have to order or steal) and can be "capped out" in their effectiveness with an upgraded thermo.

"woe, infrared singletank upon ye" is better reserved to one for that incredible tradeoff, or at least requiring you to carry them openly, drag them, drag them in a container, and potentially double-fist it if you really want to have the most Bomba possible in one run.

They're still just as potent if not more now for use as actual remote explosive/mines with this - more because people have gotten comfortable with having the singletank yeeted at their head before joining the dead as the default mode for these things. They aren't checking lockers, crates, or obvious trap hallways like they should and hopefully will if use changes to reflect this change.

Elegant adjustment for now, shouldn't be controversial. If it is to you, you should probably up your game and learn to do toxins or more advanced chem shit.

**Comment by CRITAWAKETS** [comment]
why the fuck do these exist anyway, it makes TTVs basically only just for sabotage (and sci, obviously) since the only thing you're getting is a bigger boom

make TTVs have value, remove assembly bombs

**Comment by Overuseddart** [comment]
Maybe it should work more like the improvised explosive, it might be good if you make it so you have to prime it and it randomly heats the bomb in a time from 1-12 seconds

**Comment by Jackal-boop** [comment]
i think this is an issue of people gaming too hard and reusing gimmicks. assembly bombs are cool and dont have the power of a TTV, thus they arent bulky, simply start banning people if they keep doing the assembly bomb spam every shift

**Comment by Overuseddart** [comment]
> i think this is an issue of people gaming too hard and reusing gimmicks. assembly bombs are cool and dont have the power of a TTV, thus they arent bulky, simply start banning people if they keep doing the assembly bomb spam every shift

That wouldn’t actually solve any points the pr brings up, enough players know how to make the bomb where even if a player doesn't repeat the bomb every shift, there’s a good chance there still would be one

**Comment by ExcessiveUseOfCobblestone** [comment]
> i think this is an issue of people gaming too hard and reusing gimmicks. assembly bombs are cool and dont have the power of a TTV, thus they arent bulky, simply start banning people if they keep doing the assembly bomb spam every shift

The more likely occurrence is not John Bombs making his mix every round as that's easy, it's John Bomb makes it on round 1 and Jane Bomb makes it on round 2 and Sally Bomb makes it on round 3. For the rest of the players its no different who makes it, so while you can enforce the rule it doesnt fix the issue.

The "fun" part of bombs is that it's static across rounds so unlike say an issue like boning where you can't reasonably anticipate the exact same response (which we also made difficult), bombs can be made round after round outside of events that blow up toxins and/or turn off environmentals, or players preemptively limiting the bomb maker which would also be seen as scummy.

**Comment by Singul0** [comment]
Eh, what I like most from assembly tanks is its size, you can do really sneaky stuff and tons of gimmicks (if you're creative enough) that you simply cant do with ttvs due to their sheer size and destruction.

**Comment by Pickle-Coding** [comment]
> > The problem is how easy it is to make, not how easy it is to carry. Nerf about 674 plasma which every assistant could make roundstart will affect good atmos players.
> 
> 
> 
> Nerfs based on making bomb recipes harder have a long and sordid history of working for about 2 days before the new optimal recipe becomes widely known.

Not saying their solution is good (it encourages the snowflakiness of assembly bombs), but this argument does not apply here. Assembly bombs work differently to single tanks. The optimal recipe is just defined by a series of if checks, which is basically found by codediving or brute force (boring) since there's no natural logic to it.

Their solution either implies nerfing the effects of the highest tier explosion, which is unavoidable, or increasing the requirements to reach the max tier explosion. Doing the latter could increase the requirements enough that it would require the player to put more work in order to achieve the recipe, such as burning plasma or heating with fusion.

I hate assembly bombs. Not for the balance but because of how they internally work.

**Comment by breadcrombs** [comment]
don't assembly bombs require t2 thermomachine parts to even make???

**Comment by SomeoneYouProbablyKnow** [comment]
> don't assembly bombs require t2 thermomachine parts to even make???

Yes but that is not a serious hurdle. Most maps start with those in engi storage and RnD unlocking them by minute 15

File: code/game/objects/items/tanks/tanks.dm [hunk 1]
```diff
@@ -440,6 +440,10 @@
 
 	if(LAZYLEN(assembly.assemblies) == igniter_count)
 		return
+	
+	if(isitem(loc)) // we are in a storage item
+		balloon_alert(user, "cant reach it!")
```
Comment (dragomagol) [inline code/game/objects/items/tanks/tanks.dm:445]:
```suggestion
		balloon_alert(user, "cant reach!")
```

File: code/game/objects/items/tanks/tanks.dm [hunk 2]
```diff
@@ -440,6 +440,10 @@
 
 	if(LAZYLEN(assembly.assemblies) == igniter_count)
 		return
+	
+	if(isitem(loc)) // we are in a storage item
+		balloon_alert(user, "cant reach!")
```
Comment (dragomagol) [inline code/game/objects/items/tanks/tanks.dm:445]:
```suggestion
		balloon_alert(user, "can't reach!")
```

**Comment by optimumtact** [comment]
image

**Pull request merged**